content/case-studies/imagine-learning.md

```diff
@@ -0,0 +1,135 @@
+
+## Scaling Education Infrastructure for the Next Generation
+
+As a technology company focused on K-12 education, Imagine Learning's infrastructure needs are both complex and critical. Their existing Internal Developer Platform was struggling to keep pace with evolving demands, creating several challenges:
+
+1. **Limited infrastructure visibility:** "It was really hard to determine the current state of every environment we managed," notes Blake Romano, Senior Software Engineer at Imagine Learning. Platform engineers lacked a unified view of their infrastructure, making it difficult to maintain consistency and troubleshoot issues.
+
+2. **Complex, fragmented deployment processes:** "Deploying wasn't easy; there were multiple different pipelines depending on what you were deploying and how it would go out," explains Romano. This complexity increased cognitive load on developers and slowed down delivery cycles.
+
+3. **Specialized language requirements:** Their existing infrastructure as code solution required team members to learn yet another specialized language outside their regular development workflows.
+
+4. **Manual synchronization between infrastructure and Kubernetes:** Keeping Kubernetes configurations aligned with infrastructure changes required significant manual effort, increasing the risk of configuration drift and errors.
+
+5. **Limited scalability:** As the company continued to grow, they needed a solution that could potentially scale to hundreds of environments while maintaining security and consistency.
+
+These challenges were particularly acute because of Imagine Learning's commitment to educational excellence. Any downtime or inconsistency could directly impact teachers and students who rely on their platforms daily.
+
+## Building a GitOps Bridge with Pulumi
+
+After evaluating various options, Imagine Learning implemented what Romano calls the "GitOps bridge pattern" using Pulumi, ArgoCD, and GitHub to create a modern, scalable platform.
+
+{{< youtube "1Q3XPmenthg" >}}
+
+### Implementation Details
+
+Imagine Learning's solution comprises several integrated components that work together to create a seamless workflow from infrastructure provisioning to application deployment:
+
+Imagine Learning chose Pulumi to manage their multi-region infrastructure, leveraging TypeScript instead of specialized domain-specific languages. "Pulumi allows you to write infrastructure as code in a language you already know," explains Romano. "There's no more typing of this very weird syntax that I'm not super familiar with. If I'm writing my application code and I'm moving into writing some infrastructure code, it's the same coding language and the same coding semantics that I'm used to."
```

Comment:
```suggestion
Imagine Learning chose Pulumi to manage their multi-region infrastructure on AWS, leveraging TypeScript instead of specialized domain-specific languages. "Pulumi allows you to write infrastructure as code in a language you already know," explains Romano. "There's no more typing of this very weird syntax that I'm not super familiar with. If I'm writing my application code and I'm moving into writing some infrastructure code, it's the same coding language and the same coding semantics that I'm used to."
```